### DISCUSSION TITLE
Camera: Input/Output Error - Image size not modulo 16

### DISCUSSION THREAD

**Discussion by axeljerabek** [Motion-Project/motionplus#174] [Enhancements]
### Did you read the guide?

Yes

### What is the base version number of MotionPlus being used?

0.2.x

### What was the install method?

Built from source code

### What is base architecture?

x86/x64

### What is the distro being used?

Ubuntu

### Disto version number

24.04

### Camera/Sound type(s) being used?

PI camera via libcam, Network camera with RTSP, Network camera with HTTP, Sound via ALSA

### Describe the issue/problem and steps to reproduce

During startup, motionplus has several camera errors, claiming that the image size is not modulo 16, while all cams of different type have a FullHD resolution of 1920x1080, which is true since 1080 divided by 16 results in 67.5 (not a whole number). So not really a bug, but at least annoying at some level.

Would it be possible to include the standard resolutions more seamless into the startup? I think that I have to restart the streams over Nginx-RTMP again before they get accepted my motionplus.

### Relevant MotionPlus log output (at log_level 8)

```shell
Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] netcam_open_context: Norm: Unable to open camera(Mevo): Input/output error
Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] netcam_cleanup: Norm: Shut down complete.
Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] mlp_check_szimg: Substream not available.  Image sizes not modulo 16.
Jul 24 09:15:59 [NTC][ALL][02:ml02:Mevo] mlp_retry: Retrying until successful connection with camera
```

**Comment by Mr-Dave** [comment]
> Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] mlp_check_szimg: Substream not available.  Image sizes not modulo 16.  

This is a notice to the user that some functionality is not available and why it is not available.  If the cause is removed (`Image sizes not modulo 16.`) users would have to guess as to why the substream isn't providing a lower bandwith image.  

At best this would be an enhancement request to provide a user defined substream which would also need to emit lots of messages regarding adjusting sizes.

For the other messages, this looks like something related to the camera.  These don't have anything to do with the message regarding substream.

> Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] netcam_open_context: Norm: Unable to open camera(Mevo): Input/output error
Jul 24 09:15:50 [NTC][NET][02:ml02:Mevo] netcam_cleanup: Norm: Shut down complete.
{snip}
Jul 24 09:15:59 [NTC][ALL][02:ml02:Mevo] mlp_retry: Retrying until successful connection with camera

**Comment by axeljerabek** [comment]
Yes, I also would see this as an enhancement.
All cameras stream to an Nginx-RTMP and I pick the streams up with motionplus. So the streams should actually be unified. I just don't find the reason why sometimes some cam is not picked up and others are. A stream restart directly from the camera helps in most cases though. Still, before the stream restart, I cleary see a video stream on the Nginx server when I check it with VLC.
